openml/datasets/functions.py

```diff
@@ -165,6 +165,36 @@ def _get_cache_directory(dataset: OpenMLDataset) -> str:
     return _create_cache_directory_for_id(DATASETS_CACHE_DIR_NAME, dataset.dataset_id)
 
 
+def list_qualities(verbose=False):
+    """ Return list of data qualities available.
+
+    The function performs an API call to retrieve the entire list of
+    data qualities that are available are computed on the datasets uploaded.
+
+    Parameters
+    ----------
+    verbose : bool (default=False)
+        If True, prints out the list with an index
+
+    """
+    api_call = "data/qualities/list"
+    xml_string = openml._api_calls._perform_api_call(api_call, 'get')
+    qualities = xmltodict.parse(xml_string)
+    # Minimalistic check if the XML is useful
+    if 'oml:data_qualities_list' not in qualities:
+        raise ValueError('Error in return XML, does not contain '
+                         '"oml:data_qualities_list"')
+    assert type(qualities['oml:data_qualities_list']['oml:quality']) == list
+    qualities = qualities['oml:data_qualities_list']['oml:quality']
+    if verbose:
+        header = "List of available data qualities:"
+        print(header)
+        print("=" * len(header))
+        for i, quality in enumerate(qualities):
+            print("{:>3}....{}".format(i + 1, quality))
```

Comment:
If the purpose is to help with list indexing, I would print the index and not index + 1 (i.e. 0-based and not 1-based).

Reply:
If there's a good reason to start the list at 1, let me know. It's not a necessity for me.

Reply:
Hm, could you post an example of the output? I'm right now not convinced about adding a verbosity argument to one of function and think the user can do this much better by printing the returned pandas dataframe. However, for that I think the return type argument is missing.

Reply:
I was wondering this myself. The return is a `list`, so a user just using `print(enumerate(openml.datasets.list_qualities()))` is nearly identical. 
This function's output (the str representation of the list):
```['AutoCorrelation', 'CfsSubsetEval_DecisionStumpAUC', ...]```
With verbosity (the printed messages):
```
List of available data qualities:
=================================
  1....AutoCorrelation
  2....CfsSubsetEval_DecisionStumpAUC
  ...
```
using `print(list(enumerate(openml.datasets.list_qualities())))`:
```
[(0, 'AutoCorrelation'), (1, 'CfsSubsetEval_DecisionStumpAUC'),  ...]
```




ps. @Neeratyoy  also a tip in general, `enumerate` has a `start` parameter so you could have used `enumerate(qualities, start=1)` instead and not have to manually add 1 each time.

Reply:
To be honest, in hindsight, I too find the verbosity feature quite redundant and unnecessary. Initially, I wanted to have an option of an `output_format` like other listing functions. Since this function is just a simple list of strings, I kind of force-fitted this verbosity. 

@PGijsbers @mfeurer shall I omit the verbosity feature and return a list as is or enhance the printing as @PGijsbers mentioned and add the relevant docstring for the return?

Reply:
I'm in favor of dropping it. Keeping it might start a trend where we have to add more or less trivial verbosity to other functions if users find them in one place but not another.

Reply:
Alright, makes sense.